src/world/partner/sushie.c

```diff
@@ -1,42 +1,150 @@
 extern unkPartnerStruct* D_802BFDF8_320B68;
+extern s32 D_802BFEEC;
+extern f32 D_802BFEE0;
+extern void func_802BD368_31E0D8(s32, f32, f32, f32, f32, f32);
+extern void playFX_23(s32, f32, f32, f32, s32);
+extern s32 partner_use_ability(void);
+extern void partner_kill_ability_script(void);
+extern s32 D_802BFEE4;
+extern s32 D_802BFEE8;
+extern s32 D_802BFEF0;
+extern s32 D_802BFEF4;
+extern f32 D_802BFDB0_320B20;
+extern f32 D_802BFDB4_320B24;
+extern s16 D_8010C97A;
 
-INCLUDE_ASM(s32, "world/partner/sushie", func_802BD100_31DE70);
 
-INCLUDE_ASM(s32, "world/partner/sushie", func_802BD20C_31DF7C);
+void func_802BD100_31DE70(void) {
+    Npc* partnerNPC = get_npc_unsafe(NPC_PARTNER);
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    Camera* cam =  gCameras;
+    s32 phi_v1;
 
-INCLUDE_ASM(s32, "world/partner/sushie", func_802BD368_31E0D8);
+    playerStatus->position.x = partnerNPC->pos.x;
+    playerStatus->position.y = partnerNPC->pos.y + 16.0f;
+    playerStatus->position.z = partnerNPC->pos.z;
+    playerStatus->targetYaw = partnerNPC->yaw;
+
+    if (playerStatus->spriteFacingAngle < 90.0f || playerStatus->spriteFacingAngle > 270.0f) {
+        phi_v1 = -8;
+    } else {
+        phi_v1 = 8;
+    }
+
+    playerStatus->position.z -= cos_rad(((((cam->currentYaw + playerStatus->spriteFacingAngle) - 90.0f) + phi_v1) * TAU) / 360.0f) * -4.0f;
+}
+
+void func_802BD20C_31DF7C(f32* arg0, f32* arg1) {
+    f32 temp_f22;
+    f32 phi_f20;
+    f32 temp_f2 = gPartnerActionStatus.stickY;
+    f32 temp_f0 = gPartnerActionStatus.stickX;
+    
+    D_802BFDB0_320B20 = temp_f0;
+    D_802BFDB4_320B24 = temp_f2;
+    temp_f22 = clamp_angle(atan2(0.0f, 0.0f, temp_f0, -temp_f2) + gCameras->currentYaw);
+    phi_f20 = 0.0f;
+
+    if (dist2D(0.0f, 0.0f, D_802BFDB0_320B20, -D_802BFDB4_320B24) >= 1.0) {
+        if (SQ(D_802BFDB0_320B20) + SQ(D_802BFDB4_320B24) > 3025.0f) {
+            if (D_802BFEE4) {
+                phi_f20 = 2.0f;
+            } else {
+                phi_f20 = 4.0f;
+            }
+        } else {
+            phi_f20 = 2.0f;
+        }
+    }
+
+    *arg0 = temp_f22;
+    *arg1 = phi_f20;
+}
+
+void func_802BD368_31E0D8(s32 arg0, f32 arg1, f32 arg2, f32 arg3, f32 arg4, f32 arg5) {
+    CollisionStatus* collisionStatus = &gCollisionStatus;
+    f32 sp20;
+
+    arg2 += 100.0f;
+    sp20 = 200.0f;
+    
+    if (npc_raycast_down_ahead(arg0, &arg1, &arg2, &arg3, &sp20, arg4, arg5) == 0) {
+        collisionStatus->currentFloor = -1;
+        return;
+    }
+
+    collisionStatus->currentFloor = D_8010C97A;
+    D_802BFEE0 = arg2;
+}
 
 INCLUDE_ASM(s32, "world/partner/sushie", func_802BD414_31E184);
 
-INCLUDE_ASM(s32, "world/partner/sushie", func_802BE280_31EFF0);
+s32 func_802BE280_31EFF0(s32 arg0, f32* arg1, f32* arg2, f32* arg3, f32 arg4, f32 arg5, f32* arg6) {
+    f32 sp38, sp3C, sp40, sp44, sp48, sp4C, sp50, sp54, sp58;
+    s32 colliderRayResult;
+
+    sin_cos_rad((*arg6 * TAU) / 360.0f, &sp38, &sp3C);
+    sp3C = -sp3C;
```

Comment:
Do you actually have to do this assignment, or can you directly use `-sp3C` in the function call?

Reply:
assignment seems to be required